### PR TITLE
feat(Client): add `login` method

### PR DESCRIPTION
**Pull request by megatank58**

### CODE REVIEW

File: src/client/ws/Gateway.ts [hunk 1]
```diff
@@ -34,20 +32,18 @@ export class Gateway {
...
-							}
+				this.send({
+					op: 2,
+					d: {
+						token,
+						intents: this.client.intents,
+						properties: {
+							$os: 'linux',
+							$browser: 'Discord iOS',
+							$device: 'ios'
```
Comment (twisttaan) [inline src/client/ws/Gateway.ts:43]:
```suggestion
							$os: process ? process.platform : 'ailurus',
							$browser: 'ailurus',
							$device: 'ailurus'
```

**Review by thehairy — APPROVED** [review]

File: src/client/ws/Gateway.ts (hunk 1 above)
Comment (megatank58) [reply src/client/ws/Gateway.ts:43]: These are discord platforms, would there be any effect on changing them to the library name?

Comment (twisttaan) [reply src/client/ws/Gateway.ts:43]: On Discord's analytics side of things, they would see ailurus. It has no effect, lots of libraries do this anyways.

**Review by twisttaan — APPROVED** [review]
LGTM

**Pull request merged**